src/k8s/vfs/kuberesources.virtualfs.ts

```diff
@@ -0,0 +1,178 @@
+    constructor() { }
+
+    private readonly onDidChangeFileEmitter: EventEmitter<FileChangeEvent[]> = new EventEmitter<FileChangeEvent[]>();
+
+    onDidChangeFile: Event<FileChangeEvent[]> = this.onDidChangeFileEmitter.event;
+
+    watch(_uri: Uri, _options: { recursive: boolean; excludes: string[] }): Disposable {
+        // It would be quite neat to implement this to watch for changes
+        // in the cluster and update the doc accordingly.  But that is very
+        // definitely a future enhancement thing!
+        return new Disposable(() => {});
+    }
+
+    stat(_uri: Uri): FileStat {
+        return {
+            type: FileType.File,
+            ctime: 0,
+            mtime: 0,
+            size: 65536  // These files don't seem to matter for us
+        };
+    }
+
+    readDirectory(_uri: Uri): [string, FileType][] | Thenable<[string, FileType][]> {
+        return [];
+    }
+
+    createDirectory(_uri: Uri): void | Thenable<void> {
+        // no-op
+    }
+
+    readFile(uri: Uri): Uint8Array | Thenable<Uint8Array> {
+        return this.readFileAsync(uri);
+    }
+
+    async readFileAsync(uri: Uri): Promise<Uint8Array> {
+        const content = await this.loadResource(uri);
+        return new Buffer(content, 'utf8');
+    }
+
+    async loadResource(uri: Uri): Promise<string> {
+        const query = querystring.parse(uri.query);
+
+        const outputFormat = OUTPUT_FORMAT_YAML; // k8s had a preference for outputFormat selection
+        const value = query.value as string;
+        const revision = query.revision as string | undefined;
+        const ns = query.ns as string | undefined;
+        const resourceAuthority = uri.authority;
+
+        const eer = await this.execLoadResource(resourceAuthority, ns, value, revision, outputFormat);
+        if (Errorable.failed(eer)) {
+            void vscode.window.showErrorMessage(eer.error[0]);
+            throw eer.error[0];
+        }
+
+        const er = eer.result;
+        if (CliExitData.failed(er)) {
+            const message = `Get command failed: ${CliExitData.getErrorMessage(er)}`;
+            void vscode.window.showErrorMessage(message);
+            throw message;
+        }
+
+        return er.stdout;
+    }
+
+    async execLoadResource(resourceAuthority: string, ns: string | undefined, value: string, revision: string | undefined, outputFormat: string): Promise<Errorable<CliExitData>> {
+        const nsarg = ns ? `--namespace ${ns}` : '';
+        switch (resourceAuthority) {
+            case KUBECTL_RESOURCE_AUTHORITY: {
+                const ced = await Progress.execFunctionWithProgress(`Loading ${value}...`, async () => {
+                        return await CliChannel.getInstance().executeTool(new CommandText(`oc -o ${outputFormat} ${nsarg} get ${value}`), undefined, false);
+                    });
+                return { succeeded: true, result: ced };
+            }
+            case HELM_RESOURCE_AUTHORITY: {
```

Comment:
It seems like the virtual filesystem has support for loading installed helm charts as a yaml document, but currently we don't allow opening installed helm charts from the tree view. I think it might be a good idea to allow loading helm installations.

I think it's your call whether we do this in this PR or in a future one.

Reply:
I'll probably file a new issue for this.

Another thing is `vs-kubernetes.vs-kubernetes.outputFormat` preference specifying the file format for loading a deployment config (`json` or `yaml`): with this PR we're getting rid of k8s extension dependency but still trying to read its preference value:

 https://github.com/redhat-developer/vscode-openshift-tools/blob/ef004c7c08af73d2dae90a02c6082e3eb313098f/src/explorer.ts#L555C2-L565C6

I believe we need to define and manage this preference inside `vscode-openshift-tools`